Client/Frontend/Autofill/CreditCard/CreditCardBottomSheet/CreditCardBottomSheetViewModel.swift

```diff
@@ -261,10 +264,15 @@ class CreditCardBottomSheetViewModel {
         decryptedCreditCardVal.ccExpMonth = isValidMonth ? month : originalCreditCard.ccExpMonth
 
         // Year
-        let yearVal = decryptedCreditCardVal.ccExpYear
+        var decryptedYearVal = decryptedCreditCardVal.ccExpYear
+        var originalYearVal = originalCreditCard.ccExpYear
         let currentYear = Calendar.current.component(.year, from: Date()) % 100
-        let isValidYear = (currentYear...99).contains(Int(yearVal) % 100)
-        decryptedCreditCardVal.ccExpYear = isValidYear ? yearVal : originalCreditCard.ccExpYear
+        let isValidYear = (currentYear...99).contains(Int(decryptedYearVal) % 100)
+        // We need to show only 2 digits but save full year for sync
+        let period = Int64(Date.getCurrentPeriod())
+        decryptedYearVal = decryptedYearVal < 1000 ? decryptedYearVal + period : decryptedYearVal
+        originalYearVal = originalYearVal < 1000 ? originalYearVal + period : originalYearVal
```

Comment:
Was debating adding another helper method `convertTwoDigitYear to full` but decided to keep it more raw instead of adding it to TimeConstants as it's just a one-line conversion anyways.